nncf/api/compression.py

```diff
@@ -0,0 +1,316 @@
+        """
+        return self._loss.statistics(quickly_collected_only)
+
+    def prepare_for_export(self) -> None:
+        """
+        Prepare the compressed model for deployment.
+        """
+        self._model = self.strip_model(self._model)
+
+    def export_model(self, save_path: str, *args, **kwargs) -> None:
+        """
+        Used to export the compressed model for deployment. Makes method-specific
+        preparations of the model, (e.g. removing auxiliary layers that were used
+        for the model compression), then exports the model in the specified path.
+
+        :param save_path: The path to export model.
+        :param args: Advanced export options.
+        :param kwargs: Advanced export options
+        """
+
+    def strip_model(self, model: ModelType) -> ModelType:
+        """
+        Strips auxiliary layers that were used for the model compression, as it's
+        only needed for training. The method is used before exporting the model
+        in the target format.
+
+        :param model: The compressed model.
+        :return: The stripped model.
+        """
+        return model
+
+
+class CompressionAlgorithmBuilder:
+    """
+    Determines which modifications should be made to the original model in
+    order to enable algorithm-specific compression during fine-tuning.
+    """
+
+    def __init__(self, config: NNCFConfig, should_init: bool = True):
+        """
+        Initializes internal state of the compression algorithm builder
+
+        :param config: The dictionary that contains parameters of the compression
+            method.
+        :param should_init: If False, trainable parameter initialization will be
+            skipped during building.
+        """
+        self.config = config
+        self.should_init = should_init
+
+    def apply_to(self, model: ModelType) -> ModelType:
+        """
+        Applies algorithm-specific modifications to the model.
+
+        :param model: The original uncompressed model.
+        :return: The model with additional modifications necessary to enable
+            algorithm-specific compression during fine-tuning.
+        """
+        transformation_layout = self.get_transformation_layout(model)
+        return ModelTransformer(model, transformation_layout).transform()
+
+    def build_controller(self, model: ModelType) -> CompressionAlgorithmController:
+        """
+        Builds `CompressionAlgorithmController` to handle the additional modules,
+        parameters, and hooks inserted into the model to enable algorithm-specific
+        compression.
+
+        :param model: The model with additional modifications necessary to enable
+            algorithm-specific compression during fine-tuning.
+        :return: The instance of the `CompressionAlgorithmController`.
+        """
+        return CompressionAlgorithmController(model)
+
+    def get_transformation_layout(self, model: ModelType) -> TransformationLayout:
```

Comment:
With this PR in place, NNCF for PyTorch still does not use `TransformationLayout`. The common API should not be extended with `TransformationLayout` at this point in time. It may be added later, but only along with the changes in the PyTorch part.

Reply:
In line with your comment, I have opened a ticket